src/lib/ngsiNotify/QueueWorkers.cpp

```diff
@@ -99,32 +99,41 @@ static void *workerFunc(void* pSyncQ)
                        params->resource.c_str(),
                        params->content_type.c_str()));
 
-    std::string r =  httpRequestSendWithCurl(curl, params->ip,
-                                     params->port,
-                                     params->protocol,
-                                     params->verb,
-                                     params->tenant,
-                                     params->servicePath,
-                                     params->xauthToken,
-                                     params->resource,
-                                     params->content_type,
-                                     params->content,
-                                     true,
-                                     NOTIFICATION_WAIT_MODE);
-
-    if ((r != "") && (r != "error"))
+    if (simulatedNotification)
     {
-      statisticsUpdate(NotifyContextSent, params->format);
+      LM_T(LmtNotifier, ("simulatedNotification is 'true', skipping outgoing request"));
+      __sync_fetch_and_add(&noOfSimulatedNotifications, 1);
     }
-
-    // FIXME: These counters should be incremented in the other notification modes
-    if (r != "error" && r != "")
-    {
-      QueueStatistics::incSentOK();
-    }
-    else
+    else // we'll send the notification
     {
-      QueueStatistics::incSentError();
+      std::string r =  httpRequestSendWithCurl(curl, params->ip,
+                                       params->port,
+                                       params->protocol,
+                                       params->verb,
+                                       params->tenant,
+                                       params->servicePath,
+                                       params->xauthToken,
+                                       params->resource,
+                                       params->content_type,
+                                       params->content,
+                                       true,
+                                       NOTIFICATION_WAIT_MODE);
+
+      if ((r != "") && (r != "error"))
+      {
+        statisticsUpdate(NotifyContextSent, params->format);
+      }
+
+      // FIXME: These counters should be incremented in the other notification modes
```

Comment:
A bit off-topic (as this has not been created in this PR) but maybe this is a more clear wording (to avoid confussion in the future regarding the meaning of the FIXME):

"FIXME: ok and errror counter should be incremented in the other notification modes (generalizing the concept, i.e. not as member of QueueStatistics:: which seems to be tied to just the threadpool notification mode)"